### ISSUE TITLE
(🎁) Per project 'local instance'

### ISSUE THREAD

**Issue by KotlinIsland** [lensvol/intellij-blackconnect#43]
Different projects can use different versions of black, this option should be per project (or support this as an option)

**Comment by lensvol** [comment]
Could you please clarify what you mean by that? Are you talking about per-project **blackd** connection settings or about executable path to **blackd** binary?

**Comment by KotlinIsland** [comment]
I'm talking this section:
![image](https://user-images.githubusercontent.com/65446343/189466070-84bc13f7-7a14-47b8-822d-d45d17a9c767.png)

I have different versions of black for different projects, so it doesn't make any sense for BlackConnect to use the same one for all projects.

**Comment by Zeckie** [comment]
If it will have multiple blackd instances running at the same time (for multiple open projects), each one will need it's own port. Should there be an option for the plugin to automatically choose an unused high port?

Also, if the plugin starts blackd when it loads, does it stop blackd when it unloads?

**Comment by lensvol** [comment]
Yes, the plugin tries (and in 99.9% succeeds) to stop the spawned **blackd** instance.

Regarding the automatically chosen port, it will actually be too much of a hassle. This and a couple of other reasons actually led me to abandon 'per-project' instances in favour of the single global one.
